### ISSUE TITLE
"User disapproved requested chains" error

### ISSUE THREAD

**Issue by charlesteets** [WalletConnect/WalletConnectUnity#106]
I'm getting this error when trying to make a Request from a WalletConnectSignClient:
`User disapproved requested chains`

I'm not even specifying a chain (although I've tried that as well with no success - I've tried `eip155`, `eip155:80001`, and `80001`).

```
TypedData<Domain> typedData = GetPermitTypedDefintion();
typedData.SetMessage(permit);
typedData.EnsureDomainRawValuesAreInitialised();
string json = typedData.ToJson();
string from = Address.EnsureHexPrefix();
string[] payload = new string[] { from, json };
EthSignTypedData ethTypedData = new EthSignTypedData(json);
string result = await WC.Request<EthSignTypedData, string>(topic: WC.DefaultSession.Topic, data: ethTypedData);
```

```
[RpcMethod("eth_signTypedData"), RpcRequestOptions(Clock.ONE_MINUTE, 10292)]
public class EthSignTypedData : List<string>
{
    public EthSignTypedData(params string[] typedDataParams) : base(typedDataParams)
    {
    }

    [Preserve] //Needed for JSON.NET serialization
    public EthSignTypedData()
    {
    }
}
```

**Comment by charlesteets** [comment]
After further testing, `personal_sign` seems to work perfectly, but `eth_sign`, `eth_signTypedData`, and `eth_signTypedData_v4` all give the `User disapproved requested chains` exception.

The only thing I changed between testing the different RPC calls is the three `WCEthSign`s in the below snippet. Changed all three to `WCPersonalSign`, `WCEthSignTypedData`, and `WCEthSignTypedDataV4`.  

```
WCEthSign ethSignTypedData = new WCEthSign(GetCurrentAddressInfo().Address.EnsureHexPrefix(), hexUtf8);
string result = await _wc.Request<WCEthSign, string>(topic: currentSession.Topic, data: ethSignTypedData, chainId: GetCurrentAddressInfo().ChainId);
```

**Comment by charlesteets** [comment]
I found the following github post which seems like it could possibly be related to my issue. I've tried moving `eth_sign`, `eth_signTypedData`, and `eth_signTypedData_v4` to `OptionalNamespaces` in the ConnectOptions but that did not fix the issue. I am using MetaMask as the wallet by the way



@jjzhang1 Yes, of course. That is a generic error that has nothing related to chains. The problem was that some wallets do not support some methods in the "methods" array and also some events in the 'events' array. so the solutions was to move them to the optionalMethods and optionalEvents until i got a stable config for all wallets. Something like:
 projectId: this.projectId,
      chains: [1],
      optionalChains: [1, 5, 56, 42161, 11155111, 137],
      showQrModal: true,
      methods: ["personal_sign"],
      optionalMethods: [
        "personal_sign",
        "eth_sign",
        "eth_signTransaction",
        "eth_sendTransaction",
        "eth_signTypedData",
        "eth_signTypedData_v3",
        "eth_signTypedData_v4",
        "wallet_addEthereumChain",
        "eth_requestAccounts",
        "eth_accounts",
        "wallet_switchEthereumChain",
      ],
      optionalEvents: [
        "connect",
        "chainChanged",
        "accountsChanged",
        "disconnect",
      ],
 feel free to experiment with your own config. But the main idea is to not overload the "mandatory" array with stuff you do not use

_Originally posted by @sebastiancordea97 in https://github.com/WalletConnect/walletconnect-monorepo/discussions/3412#discussioncomment-7029983_
